src/Framework/TaskPropertyInfo.cs

```diff
@@ -52,6 +54,11 @@ public TaskPropertyInfo(string name, Type typeOfParameter, bool output, bool req
         /// </summary>
         public bool Required { get; private set; }
 
+        /// <summary>
+        /// This task parameter is required (analogous to the [Required] attribute)
+        /// </summary>
+        public bool AllowEmptyString { get; private set; }
```

Comment:
I'm guessing it's no big deal to add a bool here, but we do make a lot of TaskPropertyInfos, so it might be worth checking.